### PR TITLE
Build multiple Java versions

### PR DESCRIPTION
**Pull request by bennetelli**

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -4,15 +4,20 @@ on: [push, pull_request]
 
 jobs:
   build:
-
+    strategy:
+      fail-fast: true
```
Comment (hennr) [inline .github/workflows/maven.yml:8]: if you'd make this false

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -4,15 +4,20 @@ on: [push, pull_request]
...
+      matrix:
+        jdk: [8, 11, 14, 16]
```
Comment (hennr) [inline .github/workflows/maven.yml:10]: and if you make this this 8, 11
you should see the build pass now and the java 17 support could go to the graalvm ticket then: #582

**Comment by hennr** [comment]
@bennetelli Is this rebased against the master / contains my last cassandra change?

**Comment by bennetelli** [comment]
> @bennetelli Is this rebased against the master / contains my last cassandra change?

@hennr now I did. But now there is a test failing in slack integration. I tested it locally and it's working on my machine... argh.

**Comment by bennetelli** [comment]
> > @bennetelli Is this rebased against the master / contains my last cassandra change?
> 
> @hennr now I did. But now there is a test failing in slack integration. I tested it locally and it's working on my machine... argh.

ah. the push job is green, but the pull job not. What do you think, is it ok to merge, @hennr ?
I would give it a try and merge it.

**Comment by hennr** [comment]
> > > @bennetelli Is this rebased against the master / contains my last cassandra change?
> > 
> > 
> > @hennr now I did. But now there is a test failing in slack integration. I tested it locally and it's working on my machine... argh.
> 
> ah. the push job is green, but the pull job not. What do you think, is it ok to merge, @hennr ? I would give it a try and merge it.

Well, works on my machine as well...

Agreed, merge it and take it from there if it fails regularly. It never failed on my machine so far at least.

**Comment by bennetelli** [comment]
@hennr looks like it's working: https://github.com/togglz/togglz/commit/b406a6112eee65bc7a2e99e5def9e2faac95ac85

**Comment by hennr** [comment]
That a quite pleasant piece of news 👍🏼

**Comment by hennr** [comment]
@bennetelli what's the deal with build on push and pull_requests btw.? I don't get it fully.
It will build on pull requests, and commits to master and commit to all other branches, that don't are a PR yet, right?

Bad thing is the duplictate build on PRs then, I think.
Would it be ok to build only PRs and commits to the master branch instead?

**Comment by bennetelli** [comment]
Me neither. I also tried to separate both, but that was horrible as well..

yes, probably.

that's the reason why i wanted to get rid of it in he past because of the duplicated runs.

**Comment by bennetelli** [comment]
> Would it be ok to build only PRs and commits to the master branch instead?

I think we can give it a try. You can add it if you want :)

**Comment by hennr** [comment]
ok :)
#673

**Pull request merged**